### PR TITLE
Navigation Builder shouldn't load from Cache if cache is disabled

### PR DESCRIPTION
**Pull request by blankse**
Noticed in the review of https://github.com/pimcore/pimcore/pull/13177

### CODE REVIEW

File: lib/Navigation/Builder.php [hunk 1]
```diff
@@ -154,30 +154,33 @@ public function getNavigation($activeDocument = null, $navigationRootDocument =
             $navigationRootDocument = Document::getById(1);
         }
 
-        // the cache key consists out of the ID and the class name (eg. for hardlinks) of the root document and the optional html prefix
-        $cacheKeys = ['root_id__' . $navigationRootDocument->getId(), $htmlMenuIdPrefix, get_class($navigationRootDocument)];
+        $navigation = null;
+        $cacheKeys = [];
```
Comment (jdreesen) [inline lib/Navigation/Builder.php]: I don't think this is necessary.

Comment (blankse) [reply lib/Navigation/Builder.php]: I think also. But PhpStan dont :(

Comment (jdreesen) [reply lib/Navigation/Builder.php]: PhpStan complains about `$cacheKey`, not `$cacheKeys`. Apparently, it's not smart enough to understand that it gets set in `if ($cacheEnabled) ...` so it's available inside the next `if ($cacheEnabled) ...`, too.

Comment (blankse) [reply lib/Navigation/Builder.php]: Yes i mixed up the variable name in the rush. Should fit now: https://github.com/pimcore/pimcore/pull/13183/commits/b4db0cafa61992604e954216217945cb9424138d

**Review by jdreesen — APPROVED** [review]

**Comment by aryaantony92** [comment]
@blankse Thanks!

**Pull request merged**